common/variant/README.md

```diff
@@ -335,27 +335,29 @@ The Decimal type contains a scale, but no precision. The implied precision of a
+| int64                       | `6`     | INT(64, signed)                               | 8 byte little-endian                                                                                                |
+| double                      | `7`     | DOUBLE                                        | IEEE little-endian                                                                                                  |
+| decimal4                    | `8`     | DECIMAL(precision, scale)                     | 1 byte scale in range [0, 38], followed by little-endian unscaled value (see decimal table)                         |
+| decimal8                    | `9`     | DECIMAL(precision, scale)                     | 1 byte scale in range [0, 38], followed by little-endian unscaled value (see decimal table)                         |
+| decimal16                   | `10`    | DECIMAL(precision, scale)                     | 1 byte scale in range [0, 38], followed by little-endian unscaled value (see decimal table)                         |
+| date                        | `11`    | DATE                                          | 4 byte little-endian                                                                                                |
+| timestamp                   | `12`    | TIMESTAMP(true, MICROS)                       | 8-byte little-endian                                                                                                |
+| timestamp without time zone | `13`    | TIMESTAMP(false, MICROS)                      | 8-byte little-endian                                                                                                |
+| float                       | `14`    | FLOAT                                         | IEEE little-endian                                                                                                  |
+| binary                      | `15`    | BINARY                                        | 4 byte little-endian size, followed by bytes                                                                        |
+| string                      | `16`    | STRING                                        | 4 byte little-endian size, followed by UTF-8 encoded bytes                                                          |
+| binary from metadata        | `17`    | BINARY                                        | Little-endian index into the metadata dictionary. Number of bytes is equal to the metadata `offset_size`.           |
+| string from metadata        | `18`    | STRING                                        | Little-endian index into the metadata dictionary. Number of bytes is equal to the metadata `offset_size`.           |
+| year-month interval         | `19`    | YearMonthIntervalType(start_field, end_field) | 1 byte denoting start field (1 bit) and end field (1 bit) starting at LSB followed by 4-byte little-endian value.   |
```

Comment:
What would the parquet types be for these intervals?

Also, we need to describe what the start/end fields are and how exactly they are encoded in the byte.

Reply:
I had mistakenly put in the equivalent spark types here earlier. I have removed the parquet types for now as I am investigating the parquet types.

The details about the start and end field are in a paragraph after this table in this PR.

Reply:
I ran the following Python script on a parquet table containing these interval types and found that these intervals are intrinsically stored as int/long and the type info is stored in the metadata. I'll update the table to reflect this.

```
>>> import pyarrow.parquet as pq
>>> table = pq.read_table('/home/harsh.motwani/tables/part-00000-tid-8067172485220669242-1687c1be-9e28-455a-817a-449a862b4a05-0-1-c000.snappy.parquet')
>>> table.schema
ymi0: int32 not null
ymi1: int32 not null
ymi2: int32
dti0: int64
dti1: int64
-- schema metadata --
org.apache.spark.version: '4.0.0'
org.apache.spark.sql.parquet.row.metadata: '{"type":"struct","fields":[{"' + 375

>>> table.schema.metadata
OrderedDict([(b'org.apache.spark.version', b'4.0.0'), (b'org.apache.spark.sql.parquet.row.metadata', b'{"type":"struct","fields":[{"name":"ymi0","type":"interval year to month","nullable":false,"metadata":{}},{"name":"ymi1","type":"interval year","nullable":false,"metadata":{}},{"name":"ymi2","type":"interval month","nullable":true,"metadata":{}},{"name":"dti0","type":"interval day to second","nullable":true,"metadata":{}},{"name":"dti1","type":"interval hour to minute","nullable":true,"metadata":{}}]}')])
```